R/infrastructure.R

```diff
@@ -142,6 +142,50 @@ use_travis <- function(pkg = ".") {
   invisible(TRUE)
 }
 
+#' @rdname infrastructure
+#' @section \code{use_coveralls}:
+#' Add coveralls to basic travis template to a package.
+#' @export
+use_coveralls <- function(pkg = ".") {
+  pkg <- as.package(pkg)
+
+  path <- file.path(pkg$path, ".travis.yml")
+  if (!file.exists(path)) {
+    stop(".travis.yml does not exist, please run `use_travis()` to create it", call. = FALSE)
+  }
+
+  travis_content <- readLines(file.path(pkg$path, ".travis.yml"))
+
+  if (any(grepl("coveralls()", travis_content))) {
+    stop("coveralls information already added to .travis.yml", call. = FALSE)
+  }
+
+  gh <- github_info(pkg)
+  message("Adding coveralls information into .travis.yml for ", pkg$package, ". Next: \n",
+    " * Turn on coveralls for this repo at https://coveralls.io/repos/new\n",
+    " * Add a coveralls shield to your README.md:\n",
+    "[![Coverage Status]",
+      "(https://img.shields.io/coveralls/", gh$username, "/", gh$repo, ".svg)]",
+      "(https://coveralls.io/r/", gh$username, "/", gh$repo, "?branch=master)"
+  )
+
+  install_loc <- grep("^install:$", travis_content)
```

Comment:
I'm a bit leery about automatically modifying the travis file, especially now that there are two main formats. Could you please just print out instructions? (and update them for the `language: r` style)